### ISSUE TITLE
replace zone registering from string to numeric identifier

### ISSUE THREAD

**Issue by Immemor** [Immemor/RaidCore#33]
To reduce the size of translation (See https://github.com/NielsH/RaidCore/issues/15) to do, It's should be better to register a boss's room by the numeric identifier instead of a string.

Because the string must translated, and not the numeric identifier (numeric values can be enum).

These information can retrieve from carbine API with:
- GameLib.GetCurrentZoneMap().continentZoneId 
- GameLib.GetCurrentZoneMap().parentZoneId
- GameLib.GetCurrentZoneMap().Id
- GameLib.GetCurrentZoneId()

The idea is to improve this type of line: 

``` lua
mod:RegisterRestrictZone("EpFrostAir", "Elemental Vortex Alpha", "Elemental Vortex Beta", "Elemental Vortex Delta")
mod:RegisterRestrictZone("SystemDeamons", "Halls of the Infinite Mind", "Infinite Generator Core", "Lower Infinite Generator Core")
mod:RegisterEnableZone("SystemDeamons", "Halls of the Infinite Mind", "Infinite Generator Core", "Lower Infinite Generator Core"
```

The first remark is:  GameLib.GetCurrentZoneMap().Id and  GameLib.GetCurrentZoneId() provide two different values. In the capital we can easily understand the difference. GetCurrentZoneId() is like a small piece of the map, in other words, it's like the sub zone in the Map.
So, with this observation. It's should be possible to register all boss with on value of ZoneMap.

An another detail to reduce the work load for us, is to register a start room of a boss, and not all room where a player can go. So the detection zone mechanism shouldn't disable a boss, when the combat is engaged.

So to sum up, NielsH are you agree to:
- Registering boss by { continentId, ParentMapId,  MapId } ? after Elemental pair check maybe ? With or without a list of sub zone ?
- Lock the Detection Zone feature when core is running ? that avoid to register every sub zone of the fight.

An example of registering with raw numeric value (the good one), without sub zone:
https://github.com/NielsH/RaidCore/compare/master...Immemor:zone

**Comment by NielsH** [comment]
{ contentId, ParentMapId, MapId } is fine - out of memory I don't believe we have any dynamic bosses currently, but perhaps it is worth it to have support for it anyways. For example if we 3 boss rooms where the exact same boss can spawn in one of these 3 rooms. In that case, something like this might be futureproof:
`core:NewBoss("BossName", 52, { 98, 99 }, { 100, 101 })`

Just thinking out loud - perhaps there's no need at all, but if Carbine ever adds a boss like that it would be good to be able to support it like this.

To clarify the difference between the RestrictZone and EnableZone - RestrictZone makes sure it will only be active within those subzones - EnableZone makes sure it will actually activate the module upon entering (SubZoneChanged) those areas.
This is especially helpful for System Daemons, because the UnitCreated event never makes it to RaidCore because it happens during the loading screen. So once you enter Datascape it would only activate the module after targeting the bosses manually. This was fixed by adding the EnableZone bit, in combination with the Event Objectives.

We can lock zone detection while engaged with a boss - as mentioned we can release it again upon a wipe. In /theory/ it should be fine this way, but I do think we need to be a bit careful here. We need to be certain that there's no valid zone-change happening before the WIPE event is fired. Otherwise it will fail to activate the module of course.

**Comment by olbat** [comment]
If it can help you, I've [generated](https://github.com/olbat/ws-raidcore-translator#dump-map-translation-array) a dump of the hash I used in my script to translate maps: https://github.com/olbat/ws-raidcore-translator/blob/master/maps.json .

The first entry associated to each zone name is it's id#.

**Note**: The values are coming from the [wildstar.datminer.com](http://wildstar.datminer.com/maps/10000/World) website.

**Comment by Immemor** [comment]
thanks ^^. But I have already all ID. See the branch which I pushed for the second piece of language.

And this story about subzone should be erased.

**Comment by Immemor** [comment]
Issue done for 2.0
